internal/pkg/levee/levee.go

```diff
@@ -41,32 +41,29 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	sourcesMap := pass.ResultOf[source.Analyzer].(source.ResultType)
+	funcSources := pass.ResultOf[source.Analyzer].(source.ResultType)
 	taggedFields := pass.ResultOf[fieldtags.Analyzer].(fieldtags.ResultType)
 
-	propagations := map[ssa.Node]propagation.Propagation{}
-
-	for _, sources := range sourcesMap {
+	for fn, sources := range funcSources {
+		propagations := map[*source.Source]propagation.Propagation{}
```

Comment:
Since we know the size of the resulting map, consider using `make` here.  I mean, who introduced that empty map literal initialization anyw-- oh...